VSRAD.Package/BuildTools/Errors/Parser.cs

```diff
@@ -54,33 +63,47 @@ private static Message ParseClangMessage(string header)
         }
 
         private static readonly Regex ScriptErrorRegex = new Regex(
-            @"\*(?<kind>[EW]),(?<code>[^:(]+)(?>\s\((?<file>.+):(?<line>\d+)\))?:\s(?<text>.+)", RegexOptions.Compiled);
+            @"\*(?<kind>[EW]),(?<code>[^:(]+)(?>\s\((?<file>.+)\))?:\s(?<text>.+)", RegexOptions.Compiled);
 
-        private static readonly Regex ScriptErrorLocationInTextRegex = new Regex(
-            @"(?<text>.+)\s\((?<file>.+):(?<line>\d+)\)", RegexOptions.Compiled);
+        private static readonly Regex ScriptErrorTextRegex = new Regex(
+            @"(?<text>.+)\s\((?<file>.+)\)", RegexOptions.Compiled);
 
-        private static Message ParseScriptMessage(string header)
+        private static readonly Regex ScriptErrorLocationsRegex = new Regex(
+            @"(?<file>.+):(?<line>\d+)", RegexOptions.Compiled);
+
+        private static List<Message> ParseScriptMessage(string header)
         {
             var match = ScriptErrorRegex.Match(header);
             if (!match.Success) return null;
 
             var code = match.Groups["code"].Value;
             var textAndMaybeLocation = match.Groups["text"].Value;
 
-            var message = new Message { Kind = ParseMessageKind(match.Groups["kind"].Value) };
+            var kind = ParseMessageKind(match.Groups["kind"].Value);
 
             if (!match.Groups["file"].Success)
-                match = ScriptErrorLocationInTextRegex.Match(textAndMaybeLocation);
+                match = ScriptErrorTextRegex.Match(textAndMaybeLocation);
+
+            var messages = new List<Message>();
 
             if (match.Success)
             {
-                message.SourceFile = match.Groups["file"].Value;
-                message.Line = int.Parse(match.Groups["line"].Value);
+                foreach (var source in match.Groups["file"].Value.Split(','))
+                {
+                    var message = new Message { Kind = kind };
+                    var locationMatch = ScriptErrorLocationsRegex.Match(source);
+                    message.SourceFile = locationMatch.Groups["file"].Value.Trim();
+                    message.Line = int.Parse(locationMatch.Groups["line"].Value);
+                    message.Text = code + ": " + match.Groups["text"].Value;
+                    messages.Add(message);
```

Comment:
This code will throw an error if `source` doesn't match `ScriptErrorLocationsRegex`. Example: `W,undefined: undefined reference to 'printf' (some error text)`. I would recommend you get rid of `Split` and handle commas using regular expression like `(?<file>[^,]+):(?<line>\d+)`.